common/src/main/java/org/apache/comet/parquet/TypeUtil.java

```diff
@@ -196,7 +196,9 @@ && isUnsignedIntTypeMatched(logicalTypeAnnotation, 64)) {
             || canReadAsBinaryDecimal(descriptor, sparkType)
             || sparkType == DataTypes.BinaryType
             // for uuid, since iceberg maps uuid to StringType
-            || sparkType == DataTypes.StringType) {
+            || sparkType == DataTypes.StringType
+                && descriptor.getPrimitiveType().getLogicalTypeAnnotation()
+                    instanceof LogicalTypeAnnotation.UUIDLogicalTypeAnnotation) {
```

Comment:
How can we test this?

One another question is, this is parquet uuid logical type, why it is Iceberg specified?

Reply:
+1

Reply:
> How can we test this?

I have made change on my local, and used the local build to test UUID in iceberg table. 

> this is parquet uuid logical type, why it is Iceberg specified?

I don't think Spark support UUID type. If the code comes to this line, the type has to be Iceberg UUID. If we want to be absolutely sure, we can add a flag when creating ColumnReader in Iceberg.

Reply:
Currently, this can only be tested on my local.

Reply:
@viirya If the `SparkType` is `StringType` and `LogicalTypeAnnotation` is `UUID`, then this must be iceberg UUID column, because only iceberg maps UUID to Spark `StringType`. I feel the change is safe. Or we can add an extra parameter in [getColumnReader](https://github.com/apache/arrow-datafusion-comet/blob/main/common/src/main/java/org/apache/comet/parquet/Utils.java#L39) to indicate whether the ColumnReader is an Iceberg ColumnReader.

Reply:
Okay.